core/controllers/classifier.py

```diff
@@ -96,9 +96,11 @@ class TrainedClassifierHandler(base.BaseHandler):
     @acl_decorators.open_access
     def post(self):
         """Handles POST requests."""
-        signature = self.payload.get('signature')
-        message = self.payload.get('message')
-        vm_id = self.payload.get('vm_id')
+        payload = json.loads(self.request.body)
```

Comment:
@seanlip when using `requests` module to make request, `self.payload` does not get initialized by GAE. Therefore I have to do it manually here. Do you have some idea, why it is like that?

Reply:
Actually the initialization of self.payload happens in core.controllers.base (at the bottom of `__init__`). Maybe the request dict doesn't pack everything into a payload like we do for the Oppia client/server communication? If not, it probably should...

Reply:
I didn't know that. Let me check how to fix it.

Reply:
Fixed this.